hack/testing/test-viaq-data-model.py

```diff
@@ -25,6 +25,14 @@ def empty(thing):
             return True
     return False
 
+def dictCompare(str1, str2):
+    s1_obj = json.loads(s1)
+    s2_obj = json.loads(s2)
+    if s1_obj == s2_obj:
```

Comment:
`s1_obj == s2_obj` evaluates to either `True` or `False`, so you can just do
```python
    return s1_obj == s2_obj
```

Reply:
or even simpler
```python
def dictCompare(str1, str2):
    return json.loads(s1) == json.loads(s2)
```

Reply:
Thanks!  Done.

Since the CI tests have passed at last (the last piece was the Elasitcsearch type...  If you switch the type of undefined field from json to json string or vice versa, the same field name, e.g., "undefined" cannot be used...  I guess if not yet, we want to add the note to doc.), I'd like to squash the commits.  There are plenty of your contributions and comments.  Is it ok to put them into one commit?